### PR TITLE
fix(smart-contracts): remove concept of `beneficiary`

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This deprecates and removes the concept of beneficiary from the PublicLock contract

Influence on contract size 

```
 ····························|·············|················
 |  PublicLock               ·     24.005  ·       -0.406  │
 ·---------------------------|-------------|---------------·
```

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/contracts/interfaces/IPublicLock.sol [hunk 1]
```diff
@@ -37,31 +37,19 @@ interface IPublicLock
   function publicLockVersion() external pure returns (uint16);
 
   /**
-   * @dev Called by a lock manager or beneficiary to withdraw all funds from the lock and send them to the `beneficiary`.
-   * @dev Throws if called by other than a lock manager or beneficiary
+   * @dev Called by owner to withdraw all ETH funds from the lock
```
Comment (julien51) [inline smart-contracts/contracts/interfaces/IPublicLock.sol:40]:
```suggestion
   * @dev Called by owner to withdraw all funds from the lock
```

File: smart-contracts/contracts/interfaces/IPublicLock.sol [hunk 2]
```diff
@@ -403,7 +382,7 @@ interface IPublicLock
   ///===================================================================
   /// Auto-generated getter functions from public state variables
 
-  function beneficiary() external view returns (address );
+  function beneficiary() external view returns (address);
```
Comment (julien51) [inline smart-contracts/contracts/interfaces/IPublicLock.sol:385]: Thx should go, right?

**Review by julien51 — APPROVED** [review]

File: smart-contracts/contracts/interfaces/IPublicLock.sol (hunk 2 above)
Comment (clemsos) [reply smart-contracts/contracts/interfaces/IPublicLock.sol]: true. I marked it as private so it wont appear anymore - even though we keep it for storage layout

Comment (julien51) [reply smart-contracts/contracts/interfaces/IPublicLock.sol]: Great!

**Review by julien51 — APPROVED** [review]

**Pull request merged**